pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -267,23 +269,84 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		reqLogger.Info("Created Service", "Service", serDNS)
 
 		if infinispan.IsExposed() {
-			externalService := r.serviceExternal(infinispan)
-			if externalService != nil {
-				err = r.client.Create(context.TODO(), externalService)
-				if err != nil && !errors.IsAlreadyExists(err) {
-					reqLogger.Error(err, "failed to create external Service", "Service", ser)
+			switch infinispan.Spec.Expose.Type {
+			case infinispanv1.ExposeTypeLoadBalancer, infinispanv1.ExposeTypeNodePort:
+				externalService := r.serviceExternal(infinispan)
+				if externalService != nil {
+					err = r.client.Create(context.TODO(), externalService)
+					if err != nil && !errors.IsAlreadyExists(err) {
+						reqLogger.Error(err, "failed to create external Service", "Service", ser)
+						return reconcile.Result{}, err
+					}
+					err = r.client.Update(context.TODO(), infinispan)
+					if err != nil {
+						reqLogger.Error(err, "Failed to update Infinispan with Service", "Service", ser)
+						return reconcile.Result{}, err
+					}
+					reqLogger.Info("Created External Service", "Service", externalService)
+				} else {
+					reqLogger.Info("External Service NOT created. Unsupported type?", "Service", nil)
+
+				}
+			case infinispanv1.ExposeTypeRoute:
+				lsService := ispncom.LabelsResource(infinispan.ObjectMeta.Name, "infinispan-service-external")
+				route := openshiftv1.Route{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Route",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      infinispan.Name,
+						Namespace: infinispan.Namespace,
+						Labels:    lsService,
+					},
+					Spec: openshiftv1.RouteSpec{To: openshiftv1.RouteTargetReference{Kind: "Service",
+						Name: infinispan.Name},
+						TLS: &openshiftv1.TLSConfig{Termination: openshiftv1.TLSTerminationPassthrough},
```

Comment:
@rigazilla , it would be nice to have an option to configure Host for route. We may implement it in next version.
Wdyt?

Reply:
add host configuration